webapp/src/reducers.tsx

```diff
@@ -1,17 +1,22 @@
-import {AnyAction, combineReducers} from 'redux';
+import {Reducer, combineReducers} from 'redux';
 
 import action_types from './action_types';
 
-export const openBulkInviteChannelModal = (channelId: string) => {
-    return {
-        type: action_types.OPEN_BULK_ADD_CHANNEL_MODAL,
-        data: {
-            channelId,
-        },
-    };
+export type PluginAction = {
+    type: string;
+    data?: object;
 };
 
-const bulkAddChannelModalVisible = (state = false, action: AnyAction) => {
+type BulkAddChannelModalActionData = {
+    channelId?: string;
+};
+
+export type BulkAddChannelModalAction = {
+    type: string;
+    data?: BulkAddChannelModalActionData;
```

Comment:
Another way of organizing this would be:

```ts
export type PluginAction<T=undefined> = {
    type: string;
    data?: T;
};

export type BulkAddChannelModalAction = PluginAction<BulkAddChannelModalActionData>;